### ISSUE TITLE
[Problem] Expansion and building is punished

### ISSUE THREAD

**Issue by Gireen** [Unvanquished/gameplay#28]
Currently the best strategy is to build as few buildings as possible and no outposts.
If you expand and build outposts with miners, the construction points used turn into massive debt when the buildings are destroyed. 
A team is then no longer able to build something and starts spamming miners to reduce the debt.

**Disclaimer**
>All problems issues are treated as opinions until agree upon

**Comment by Viech** [comment]
I'm not happy with negative buildpoints as a concept. I would say I stopped experimenting at a suboptimal point.
On the other hand it is very easy to build a system where mindless spamming of stuff is not punished at all. We had a system earler where one could mine a lot of BP like this and not lose them when the fwd gets destroyed. A better system needs to find a middle ground.

**Comment by bmorel** [comment]
I suppose the traditional RTS way would work: you mine and *store* the mining's content in structures, mining structures have a small storage pool and there are dedicated storage structures.
If a storing structure is destroyed, resource is either redistributed automatically for free and surplus lost, or the content is just lost, depends on the games.

**Comment by SeanScherer** [comment]
I just played my first game (vs bots, so take this with a grain of salt - oh, and Tremulous background :P ).

I was playing builder a bit, and experimenting with expanding with drills - and of course ended up with the problem that I couldn't even rebuild the Armory when it was destroyed (or - had to rebuild a bunch of drills first, which led to further problems down the road of course :P ). At least it seemed rather unituitive to me (but then of course, like I said, I'm coming from Trem).

**Comment by bmorel** [comment]
you're welcome, this is a good enough place to talk about that kind of problems

**Comment by SeanScherer** [comment]
Hi, thanks :).
I just read up that destroyed buildings don't insta-give back their build points (as it was in Tremulous). I guess together with the drill-mechanics (I'm afraid I still don't fully understand), this got me confused :P. 

(oh, and I didn't see this was an older issue - I can open new if prefered!).

**Comment by bmorel** [comment]
I'd rather keep the number of issues followable. Better to talk, than to ignore old stuff. To me, you did the right thing.

**Comment by bmorel** [comment]
> If you expand and build outposts with miners, the construction points used turn into massive debt when the buildings are destroyed.

There is another side on this issue which has not been talked about here:

When you build a forward base, it's very fragile, especially for aliens. The most usual results is, it gets destroyed, giving enemy team momentum while your own team loose momentum.
This 2nd aspect of the punishment is even harder for a team than the BP debt, which can be worked around.

**Comment by DolceTriade** [comment]
We want to encourage building fwds, but also reward killing structures. I think it could be interesting to give momentum based on how quickly a group of buildings gets destroyed. I think that is quite suitable for the concept of "momentum", where the team killing lots of structures should gain momentum and be rewarded for it.

**Comment by bmorel** [comment]
This would imply that destroying a structure or killing someone gives a momentum bonus that quickly fades, if you destroy something after that. This is not the current mechanism at all.

**Comment by bmorel** [comment]
2 players said it would be good to have rewards to kill buildings, in credits, none against.

**Comment by DolceTriade** [comment]
Killing structures giving credits still punishes forwards right?

**Comment by bmorel** [comment]
Yes, but I could not remember where this idea (reward killing buildings) was submitted, so to avoid this info to be lost, I may have hastily put it here (before forgetting).

**Comment by DolceTriade** [comment]
I vehemently disagree. Credits for killing structures means that people will be more reluctant to build forwards because that means the enemy can get evos/credits from killing them.

Your core premise for the problem is absolutely correct: we punish building forwards. Building forwards is a high risk/high reward venture between giving the enemy team more momentum, getting your own BP queued, etc. We need a different system.

The idea behind momentum is simple (albeit opaque): reward players (and the team) for doing things that help the team. This includes but is not limited to:
 - Building
 - Killing enemies
 - Killing enemy structures

And if you stop performing these activities, then you will be consequently punished (by losing momentum). Ultimately, we need gameplay tests to verify our hypotheses, so we'll need
a) someone to code up a test (should be trivial for this)
b) coordinate a dev game (we should bring this back anyways).

What do you say? Let's just put this to the test. If you want, I can help code it to prototype it.

**Comment by bmorel** [comment]
> I vehemently disagree

No problem. I was basically reporting things players said (on 5 players, 2 answered), even if *I* asked about 2 things:

1. rewarding destruction of buildings, as it was said more than once, maybe only on IRC, that aliens lack an incentive to attack bunkers
2. rewards for partial damages (I think this idea came from @Gireen ) but had no answer for this one.

My point was more to collect opinions than to express my own here.

> b) coordinate a dev game (we should bring this back anyways).

This one seems hard :/

> What do you say? Let's just put this to the test. If you want, I can help code it to prototype it.

This would be very helpful to do experiments on that topic, planned for 0.54 or after, so that we have time to properly test and talk about them, with players participating, ideally.

Note that I agree with what you said, that it would reward too much enemy team.

**Comment by bmorel** [comment]
Oh, while I am at it. Doing a different post *on purpose*, to keep things readable.

I was thinking few days ago: when you build, your team *instantly* loose the build's BP, and if building is destroyed at, said, 10% of the process, you still loose 100% of the BP.
It would be a pretty obvious way to reduce the punishment to have 90% of the cost back (in that example). This could apply to momentum earning/loss or whatever other rewards we decide to pack in there.

**Comment by Gireen** [comment]
@bmorel please stay on topic. you can create separate issues for other problems/ideas or also experiences. 

This issue is about the negative BP. Not reward for kills, momentum or BP lost trough destruction of partly constructed buildings.

**Comment by bmorel** [comment]
I disagree that BP is the only punition.
